movement/roi/base.py

```diff
@@ -200,3 +313,257 @@ def contains_point(
+        <movement.utils.vector.compute_signed_angle_2d>` between the approach
+        vector and a given reference vector.
+
+        Parameters
+        ----------
+        position : xarray.DataArray
+            ``DataArray`` of spatial positions.
+        angle_rotates : Literal["approach to ref", "ref to approach"]
+            Direction of the signed angle returned. Default is
+            ``"approach to ref"``.
+        boundary_only : bool
+            If ``True``, the allocentric angle to the closest boundary point of
+            the region is computed. Default ``False``.
+        in_degrees : bool
+            If ``True``, angles are returned in degrees. Otherwise angles are
+            returned in radians. Default ``False``.
+        reference_vector : ArrayLike | xr.DataArray
+            The reference vector to be used. Dimensions must be compatible with
+            the argument of the same name that is passed to
+            :func:`compute_signed_angle_2d`. Default ``(1., 0.)``.
+
+        See Also
+        --------
+        compute_approach_vector :
+            The method used to compute the approach vector.
+        compute_egocentric_angle_to_nearest_point :
+            Related class method for computing the egocentric angle to the
+            region.
+        movement.utils.vector.compute_signed_angle_2d :
+            The underlying function used to compute the signed angle between
+            the approach vector and the reference vector.
+
+        """
+        if reference_vector is None:
+            reference_vector = np.array([[1.0, 0.0]])
+
+        return self._boundary_angle_computation(
+            position=position,
+            reference_vector=reference_vector,
+            how_to_compute_vector_to_region=lambda p: self._reassign_space_dim(
+                self.compute_approach_vector(
+                    p, boundary_only=boundary_only, unit=False
+                ),
+                "vector to",
+            ),
+            angle_rotates=angle_rotates.replace("approach", "vec"),  # type: ignore
+            in_degrees=in_degrees,
+        )
+
+    def compute_egocentric_angle_to_nearest_point(
+        self,
+        direction: xr.DataArray,
+        position: xr.DataArray,
+        angle_rotates: Literal[
+            "approach to direction", "direction to approach"
+        ] = "approach to direction",
+        boundary_only: bool = False,
+        in_degrees: bool = False,
+    ) -> xr.DataArray:
+        """Compute the egocentric angle to the nearest point in the region.
+
+        With the term "egocentric", we indicate that we are measuring angles
+        with respect to a reference frame that is varying in time relative to
+        the experimental/camera setup.
+
+        The egocentric angle is the signed angle between the approach vector
+        and a ``direction`` vector (examples include the forward vector of a
+        given individual or keypoint).
```

Comment:
It's hard for me to image the "forward vector" for a keypoint, at least with our definition of that term.
I've supplied an alternative example.
```suggestion
        and a ``direction`` vector (examples include the forward vector of
        a given individual, or the velocity vector of a given point).
```